.github/workflows/pull-request-checks.yml

```diff
@@ -0,0 +1,52 @@
+name: "Pull request checks"
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
+
+# only run this once per PR at a time
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number }}
+  cancel-in-progress: true
+
+jobs:
+  code-quality:
+    uses: ./.github/workflows/code-quality.yml
+    with:
+      branch: ${{ github.event.pull_request.head.ref }}
+      repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+  unit-tests:
+    uses: ./.github/workflows/unit-tests.yml
+    with:
+      package: ${{ matrix.package }}
+      branch: ${{ github.event.pull_request.head.ref }}
+      repository: ${{ github.event.pull_request.head.repo.full_name }}
+    strategy:
+      matrix:
+        package: ["dbt-athena", "dbt-athena-community"]
+
+  integration-tests:
+    if: contains(github.event.pull_request.labels.*.name, 'ci:integration-tests')
```

Comment:
Do we need to gate running integration test behind a label?

Reply:
That's what the repo originally had, and I don't hate it. What might also work is only running the workflow if the PR is in Ready for Review, but holding off if it's in Draft status. I think one of the reasons this existed is because some of the integration tests cannot run in parallel, which is why we also only run integration tests for one version of python. We should eventually figure out how to fix that. But in the meantime, if we have two PRs with changes happening in parallel, or scheduled tests running, etc. we could easily collide. This gives us a little more control. I can also remove it for now and address it separately.

Reply:
For context:

the reason why integration tests were under a label was simply because of cost - running integration tests concurrently was never an issue, because each test suite created a dedicated DB.

We didn't want to have the CI actions running when the PRs were in a draft or really early stage.
Generally after a first review, and then PR reached a "almost ready" to merge, me or another maintainer added the label to trigger the integration tests.

Reply:
Thanks for the context @nicor88! Controlling cost makes sense. We don't have a control like that on other adapters, but maybe we should. I do like the more automatic nature of it depending on Draft vs Ready. @colin-rogers-dbt if you have concerns over the manual nature of adding the label, I can pull that for now and then we can address it separately if we want. Does that work?

> running integration tests concurrently was never an issue, because each test suite created a dedicated DB

Hmm, this is good to know. There's a single test that seems to always fail if I run on two versions of python in parallel. Perhaps CI is creating a DB per run, but not per matrix entry. It's probably worth looking at in a separate ticket.

Reply:
I removed the label requirement.